src/game/shell.c

```diff
@@ -182,6 +187,78 @@ void Shell_Wait(int nticks)
     }
 }
 
+void Shell_GetScreenshotName(char *str)
+{
+    // Get level title of unknown length
+    char level_title[100];
+    sprintf(level_title, "%s", g_GameFlow.levels[g_CurrentLevel].level_title);
+
+    // Trim level titles that are too long
+    if (strlen(level_title) >= LEVEL_TITLE_SIZE) {
+        level_title[LEVEL_TITLE_SIZE] = '\0';
+    }
+
+    // Prepare level title for screenshot
+    char *check = level_title;
+    bool prev_us = true; // '_' after timestamp before title
+    int idx = 0;
+
+    while (*check != '\0') {
+        if ((*check < 'A' || *check > 'Z') && (*check < 'a' || *check > 'z')
+            && (*check < '0' || *check > '9') && *check != '\'' && *check != '.'
+            && *check != ' ') {
+            // Strip non alphanumeric chars
+            memmove(
+                &level_title[idx], &level_title[idx + 1],
+                strlen(level_title) - idx);
+        } else if (*check == ' ') {
+            // Replace spaces with a single underscore
+            if (prev_us) {
+                memmove(
+                    &level_title[idx], &level_title[idx + 1],
+                    strlen(level_title) - idx);
+            } else {
+                *check++ = '_';
+                idx++;
+                prev_us = true;
+            }
+        } else if (*check == '\'' || *check == '.') {
+            // Strip ' and .
+            memmove(
+                &level_title[idx], &level_title[idx + 1],
+                strlen(level_title) - idx);
+        } else {
+            *check++;
+            idx++;
+            prev_us = false;
+        }
+    }
+
+    // If title totally invalid, name it based on level number
+    if (strlen(level_title) == 0) {
+        char new_title[LEVEL_TITLE_SIZE];
+        sprintf(new_title, "Level_%d", g_CurrentLevel);
+        memmove(level_title, new_title, strlen(new_title));
```

Comment:
I think `strcpy`  here would be more readable

Reply:
Ok. My C is super rusty, and I try to be super cautious lol. Thanks for your patience on this.